pyface/ui/qt4/workbench/split_tab_widget.py

```diff
@@ -49,9 +49,7 @@ def __init__(self, *args):
 
         self.clear()
 
-        QtCore.QObject.connect(QtGui.QApplication.instance(),
-                QtCore.SIGNAL('focusChanged(QWidget *,QWidget *)'),
-                self._focus_changed)
+        QtGui.QApplication.instance().focusChanged.connect(self._focus_changed)
```

Comment:
I don't know if it handles `None` values gracefully, but I think things would have to be going very wrong for there to be no `QApplication` at the point where the `__init__` is being called.

From the point of view of testing, we should probably be a little worried about this signal handler potentially hanging around across test boundaries.